src/test/app/LedgerLoad_test.cpp

```diff
@@ -199,6 +233,99 @@ class LedgerLoad_test : public beast::unit_test::suite
             sd.ledger[jss::ledger][jss::accountState].size());
     }
 
+    void
+    testReplay(SetupData const& sd)
+    {
+        testcase("Load and replay by hash");
+        using namespace test::jtx;
+
+        // create a new env with the ledger hash specified for startup
+        auto ledgerHash = to_string(sd.hashes[sd.hashes.size() - 1]);
+        boost::erase_all(ledgerHash, "\"");
+        Env env(
+            *this,
+            envconfig(
+                ledgerConfig,
+                sd.dbPath,
+                ledgerHash,
+                Config::REPLAY,
+                std::nullopt),
+            nullptr,
+            beast::severities::kDisabled);
+        auto const jrb = env.rpc("ledger", "current", "full")[jss::result];
+        BEAST_EXPECT(jrb[jss::ledger][jss::accountState].size() == 97);
+
+        env.close();
+        auto const closed = env.rpc("ledger", "current", "full")[jss::result];
+        BEAST_EXPECT(closed[jss::ledger][jss::accountState].size() == 98);
```

Comment:
What is the meaning of the tests for 97, 98, "<=", etc? Can you add some comments to clarify. Just adding the comments in one code block would be useful.

Reply:
Added comment to explain - in `replay` mode, the ledger being replayed needs to be explicitly accepted e.g. with `env.close()` . Before this call, we are getting an earlier ledger as `current`, with one fewer state objects.